### PR TITLE
[feat:#198][tsdb]:build data family based on timestamp

### PR DESCRIPTION
**Pull request by stone1100**

### CODE REVIEW

File: tsdb/family.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"github.com/lindb/lindb/kv"
+	"github.com/lindb/lindb/tsdb/series"
+)
+
+// DataFamily represents a storage unit for time series data, support multi-version.
+type DataFamily interface {
+	// Scan scans time series data based on query condition, returns scan iterator
+	Scan(scanContext *series.ScanContext) series.ScanIterator
```
Comment (CodingCrush) [inline tsdb/family.go]: use series.Scanner ?

File: tsdb/segment.go [hunk 2]
```diff
@@ -137,6 +145,8 @@ type Segment interface {
 	BaseTime() int64
 	// GetDataFamilyScanners returns data family list by time range, return nil if not match
 	GetDataFamilyScanners(timeRange timeutil.TimeRange) []series.DataFamilyScanner
```
Comment (CodingCrush) [inline tsdb/segment.go]: are these 2 methods redundant?

Comment (stone1100) [reply tsdb/segment.go]: segment invokes getScanner method for family level scanning

File: tsdb/family.go (hunk 1 above)
Comment (stone1100) [reply tsdb/family.go]: i think can remove series.Scanner, use family.getScanner() directly

**Pull request merged**